builtin/logical/pki/path_fetch_keys.go

```diff
@@ -0,0 +1,225 @@
+		HelpDescription: pathListKeysHelpDesc,
+	}
+}
+
+const (
+	pathListKeysHelpSyn  = `Fetch a list of all issuer keys`
+	pathListKeysHelpDesc = `This endpoint allows listing of known backing keys, returning
+their identifier and their name (if set).`
+)
+
+func (b *backend) pathListKeysHandler(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var responseKeys []string
+	responseInfo := make(map[string]interface{})
+
+	entries, err := listKeys(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, identifier := range entries {
+		key, err := fetchKeyById(ctx, req.Storage, identifier)
+		if err != nil {
+			return nil, err
+		}
+
+		responseKeys = append(responseKeys, string(identifier))
+		responseInfo[string(identifier)] = map[string]interface{}{
+			keyNameParam: key.Name,
+		}
+
+	}
+	return logical.ListResponseWithInfo(responseKeys, responseInfo), nil
+}
+
+func pathKey(b *backend) *framework.Path {
+	pattern := "key/" + framework.GenericNameRegex(keyRefParam)
+	return buildPathKey(b, pattern)
+}
+
+func buildPathKey(b *backend, pattern string) *framework.Path {
+	return &framework.Path{
+		Pattern: pattern,
+
+		Fields: map[string]*framework.FieldSchema{
+			keyRefParam: {
+				Type:        framework.TypeString,
+				Description: `Reference to key; either "default" for the configured default key, an identifier of a key, or the name assigned to the key.`,
+				Default:     "default",
+			},
+			keyNameParam: {
+				Type:        framework.TypeString,
+				Description: `Human-readable name for this key.`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback:                    b.pathGetKeyHandler,
+				ForwardPerformanceStandby:   false,
+				ForwardPerformanceSecondary: false,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathUpdateKeyHandler,
+				ForwardPerformanceStandby:   true,
+				ForwardPerformanceSecondary: true,
+			},
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback:                    b.pathDeleteKeyHandler,
+				ForwardPerformanceStandby:   true,
+				ForwardPerformanceSecondary: true,
+			},
+		},
+
+		HelpSynopsis:    pathKeysHelpSyn,
+		HelpDescription: pathKeysHelpDesc,
+	}
+}
+
+const (
+	pathKeysHelpSyn  = `Fetch a single issuer key`
+	pathKeysHelpDesc = `This allows fetching information associated with the underlying key.
+
+:ref can be either the literal value "default", in which case /config/keys
+will be consulted for the present default key, an identifier of a key,
+or its assigned name value.
+
+Writing to /key/:ref allows updating of the name field associated with
+the certificate.
+`
+)
+
+func (b *backend) pathGetKeyHandler(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	keyRef := data.Get("ref").(string)
```

Comment:
Needs to be `keyRefParam` as otherwise this panics :-) 

```
2022-04-27T18:09:50.747-0400 [INFO]  http: panic serving 127.0.0.1:36134: field ref not in the schema
```

Reply:
That's what I get for not testing after making the field name changes.  Sigh.  Fixed, thank you.